src/doc_builder/autodoc.py

```diff
@@ -255,6 +256,36 @@ def is_rst_docstring(docstring):
     return False
 
 
+# Re pattern to catch example introduction & example code block.
+_re_example_codeblock = re.compile(r"((.*(E|e)xample.*:\s+)?^\s*```(((?!```)(.|\n))*)+```)", re.MULTILINE)
+
+
+def hashlink_example_codeblock(object_doc, object_anchor):
+    """
+    Returns the svelte `ExampleCodeBlock` component string.
+
+    Args:
+    - **object_doc** (`str`) -- The docstring of the the object.
+    - **anchor** (`str`) -- The anchor name of the function or class that will be used for hash links.
+    """
+
+    def svelte_generator():
+        id = 1
+        while True:
+            id_str = "" if id == 1 else f"-{id}"
+            example_anchor = f"{object_anchor}.example{id_str}"
+            match = yield
+            yield f'<ExampleCodeBlock anchor="{example_anchor}">\n\n{match.group(1)}\n\n</ExampleCodeBlock>'
+            id += 1
```

Comment:
I don't think you need to do this complex generator if you just define `id` (which should be `example_id`) before this function and then add a `nonlocal example_id` inside of it.